File: backend/app/schemas/market.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class MarketRegimeResponse(BaseModel):
    id: UUID
    symbol: str
    start_at: datetime
    end_at: datetime | None
    regime_type: str
    confidence: float
    features: dict | None

    model_config = {"from_attributes": True}

File: frontend/src/lib/api/market.ts
import { api } from '@/lib/api/client'

export interface MarketRegime {
  id: string
  symbol: string
  start_at: string
  end_at: string | null
  regime_type: string
  confidence: number
  features: Record<string, unknown> | null
}

export async function getMarketRegimes(symbol = 'BTC-USD'): Promise<MarketRegime[]> {
  return api.get(`/api/v1/market/regimes?symbol=${symbol}`)
}
